extension/src/background/helpers/dataStorage.ts

```diff
@@ -148,17 +152,17 @@ export const migrateSorobanRpcUrlNetworkDetails = async () => {
 // This migration migrates the storage for custom tokens IDs to be keyed by network
 const migrateTokenIdList = async () => {
   const localStore = dataStorageAccess(browserLocalStorage);
-  const tokenIdsByKey = (await localStore.getItem(TOKEN_ID_LIST)) as Record<
-    string,
-    object
-  >;
+  const tokenIdsByKey = await localStore.getItem(TOKEN_ID_LIST);
   const storageVersion = (await localStore.getItem(STORAGE_VERSION)) as string;
 
   if (!storageVersion || semver.lt(storageVersion, "1.0.0")) {
-    const newTokenList = {
-      [NETWORKS.FUTURENET]: tokenIdsByKey,
-    };
-    await localStore.setItem(TOKEN_ID_LIST, newTokenList);
+    if (Array.isArray(tokenIdsByKey)) {
```

Comment:
Now that the bug is fixed, there shouldn't be a scenario where a user has the new schema for tokenIdList but enters into this migration anyway.

I'm just adding this extra check to make sure in the event this bug comes back, that we don't mutate the schema if it's already correct